Refactor `DateTimeFormatter` Filter (#160)
laminas/laminas-filter: docs/book/v3/standard-filters.md

methods for `target` are `getTarget()` and `setTarget()`. You can also use the `setOptions()` method
which accepts an array of all options.

## DateTimeFormatter

The filter normalize a datetime string to match the filter specified format

### Supported Options

> There are no additional options for `Laminas\Filter\Digits`

Reviewer: Wrong class name - also, options _are_ supported and they will need documenting

### Basic Usage

```php
$filter = new \Laminas\Filter\DateTimeFormatter();

echo $filter->filter('2024-01-01'); // => 2024-01-01T00:00:00+00:00
echo $filter->filter(1_359_739_801); // => 2013-02-01T17:30:01+00:00
echo $filter->filter(new DateTimeImmutable('2024-01-01')) // => 2024-01-01T00:00:00+00:00 

> $filter->setFormat('d-m-Y');

Reviewer: ```suggestion
$filter = new \Laminas\Filter\DateTimeFormatter(['format' => 'd-m-Y']);
```

methods for `target` are `getTarget()` and `setTarget()`. You can also use the `setOptions()` method
which accepts an array of all options.

## DateTimeFormatter

> The filter normalize a datetime string to match the filter specified format

Reviewer: ```suggestion
This filter formats either a `DateTimeInterface` object, a string, or integer that `DateTime` will understand to a date and/or time string in the configured format.
```

### Supported Options

The following options are supported for `Laminas\Filter\DateTimeFormatter`

> - `format`: a valid date format to use when formatting a string

Reviewer: ```suggestion
- `format`: a valid [date format](https://www.php.net/manual/datetime.format.php) to use when formatting a string, for example `l jS F Y`. This option defaults to `DateTimeInterface::ATOM` when unspecified
```

> - `timezone` : a valid timezone to be used when handling the date

Reviewer: ```suggestion
- `timezone`: The default timezone to apply when converting a string or integer argument to a `DateTime` instance. This option falls back to the system timezone when unspecified
```

### Basic Usage

```php
$filter = new \Laminas\Filter\DateTimeFormatter();

echo $filter->filter('2024-01-01'); // => 2024-01-01T00:00:00+00:00
echo $filter->filter(1_359_739_801); // => 2013-02-01T17:30:01+00:00
echo $filter->filter(new DateTimeImmutable('2024-01-01')) // => 2024-01-01T00:00:00+00:00 

$filter = new \Laminas\Filter\DateTimeFormatter([
    'format' => 'd-m-Y'
]);
echo $filter->filter('2024-08-16 00:00:00'); // => 16-08-2024

$filter = new \Laminas\Filter\DateTimeFormatter([
    'timezone' => 'Europe/Paris'
]);
echo $filter->filter('2024-01-01'); // => 2024-01-01T00:00:00+01:00

> ```

Reviewer 2: The code example must be split into three separate examples:

- without any options
- with format option
- with timezone option

Each example has its own heading. This improves readability.

## DateTimeFormatter

This filter formats either a `DateTimeInterface` object, a string, or integer that `DateTime` will understand to a date
and/or time string in the configured format.

### Supported Options

The following options are supported for `Laminas\Filter\DateTimeFormatter`

- `format`: a valid [date format](https://www.php.net/manual/datetime.format.php) to use when formatting a string, for example `l jS F Y`. This option defaults to `DateTimeInterface::ATOM` when unspecified
- `timezone` : The default timezone to apply when converting a string or integer argument to a `DateTime` instance. This option falls back to the system timezone when unspecified

### Basic Usage

> #### Without any options

Reviewer: ```suggestion
#### Without Any Options
```

Titles should be title case 👍

Reviewer 3: there is actually a Website which Formats any text. 
got that from @froschdesign a few years back: https://capitalizemytitle.com/style/Chicago/

```php
$filter = new \Laminas\Filter\DateTimeFormatter();

echo $filter->filter('2024-01-01'); // => 2024-01-01T00:00:00+00:00
echo $filter->filter(1_359_739_801); // => 2013-02-01T17:30:01+00:00
echo $filter->filter(new DateTimeImmutable('2024-01-01')) // => 2024-01-01T00:00:00+00:00 
```

> #### With format option

Reviewer: ```suggestion
#### With `format` Option
```

```php
$filter = new \Laminas\Filter\DateTimeFormatter([
    'format' => 'd-m-Y'
]);
echo $filter->filter('2024-08-16 00:00:00'); // => 16-08-2024
```

> #### With timezone option

Reviewer: ```suggestion
#### With `timezone` Option
```
